book: update concepts section (#331)
KDAB/cxx-qt: book/src/concepts/types.md

# Types

> CXX-Qt any types supported by CXX, these can be used in properties, invokables, and signals.

Reviewer: ```suggestion
CXX-Qt supports all types supported by CXX. These can be used in properties, invokables, and signals.
```

Author: done

> Note that there is missing support for a few CXX types https://github.com/KDAB/cxx-qt/issues/328

Author: Done

## `cxx-qt-lib` Types

> The `cxx-qt-lib` crate provides common types from Qt as CXX supported types.

Reviewer: ```suggestion
The `cxx-qt-lib` crate provides CXX bindings for common Qt types.
```

Author: Done

Use the [`cxx-qt-lib` Docs](https://docs.rs/cxx-qt-lib/latest/cxx_qt_lib/) to explore the available types.

## Defining a Custom Type

Any types that are valid CXX types should be usable with CXX-Qt as well.

> Note that the same rules apply as CXX, so a type must be trivial to pass around as value.

Reviewer: ```suggestion
Note that the same rules apply as CXX, so a type must be [trivial](https://cxx.rs/extern-c++.html?highlight=trivial#integrating-with-bindgen-generated-or-handwritten-unsafe-bindings) to pass by value.
```

Author: Done

> If they are opaque then references or pointers must be used.

Author: Done

> For examples of how to wrap Qt objects explore the `cxx-qt-lib` source code.

Reviewer: ```suggestion
For examples of how to wrap Qt objects, explore the [`cxx-qt-lib` source code](https://github.com/KDAB/cxx-qt/tree/main/crates/cxx-qt-lib).
```

Author: Done

## Opaque Type Conversions

> If your type can't be marked as trivial for CXX, but the Qt API needs to take the type by value (eg `T` rather than `const T&` or `::std::unique_ptr<T>`).

Reviewer: ```suggestion
If your type can't be marked as trivial for CXX, but the Qt API needs to take the type by value (eg `T` rather than `const T&` or `::std::unique_ptr<T>`),
```
not a complete sentence

Author: Done

Then you can define [type conversions](type-conversions.md) in C++ that convert between pointers / references and values.
Then when using properties, invokables, or signals an attribute in the macro can be used.

### Properties

> In the following example using `cxx_type` means that `OpaqueExampleType` will be the type of the `Q_PROPERTY`, getter, and setter in C++.

Reviewer: I'm confused. There's no `cxx_type` in the example code.

Author: In the `cxx_type = "OpaqueExampleType" ` in the `#[qproperty(cxx_type = "OpaqueExampleType")]` line
